elfo-core/src/init.rs

```diff
@@ -29,48 +29,50 @@ use crate::{
 
 type Result<T, E = StartError> = std::result::Result<T, E>;
 
-async fn start_entrypoints(ctx: &Context, topology: &Topology) -> Result<()> {
+async fn start_entrypoints(ctx: &Context, topology: &Topology, is_check_only: bool) -> Result<()> {
     let futures = topology
         .actor_groups()
         .filter(|group| group.is_entrypoint)
-        .map(|group| {
-            let config = Default::default();
-            ctx.request_to(group.addr, UpdateConfig { config })
-                .resolve()
-                .or_else(|err| async move {
-                    match err {
-                        RequestError::Ignored => Ok(Ok(())),
-                        _ => Err(StartError::Other(
+        .map(|group| async move {
+            fn convert_response<R>(
+                group: String,
+                response: Result<Result<(), impl Into<StartErrorReason>>, RequestError<R>>,
+            ) -> Result<()> {
+                match response {
+                    Ok(Ok(())) => Ok(()),
+                    Ok(Err(e)) => Err(StartError {
+                        group,
+                        reason: e.into(),
+                    }),
+                    Err(RequestError::Ignored) => Ok(()),
+                    Err(RequestError::Closed(..)) => Err(StartError {
+                        group,
+                        reason: StartErrorReason::Other(
                             "initial messages cannot be delivered".into(),
-                        )),
-                    }
-                })
-        });
-
-    let error_count = futures::future::try_join_all(futures)
-        .await?
-        .into_iter()
-        .filter_map(Result::err)
-        .count();
-
-    if error_count == 0 {
-        Ok(())
-    } else {
-        Err(StartError::InvalidConfig)
-    }
-}
+                        ),
+                    }),
+                }
+            }
 
-async fn wait_entrypoints(ctx: &Context, topology: &Topology) -> Result<()> {
-    let futures = topology
-        .actor_groups()
-        .filter(|group| group.is_entrypoint)
-        .map(|group| ctx.request_to(group.addr, Ping).resolve());
+            let response = ctx
+                .request_to(
+                    group.addr,
+                    UpdateConfig {
+                        config: Default::default(),
+                    },
+                )
+                .resolve()
+                .await;
+            convert_response(group.name.clone(), response)?;
 
-    futures::future::try_join_all(futures)
-        .await
-        .map_err(|_| StartError::Other("entrypoint cannot start".into()))?;
+            let response = ctx
+                .request_to(group.addr, StartEntrypoint { is_check_only })
+                .resolve()
+                .await;
+            convert_response(group.name, response)
+        });
 
-    Ok(())
+    futures::future::try_join_all(futures).await.map(|_| ())
```

Comment:
One thing I initially missed is that `try_join_all` will actually return only one error. So if there were multiple entrypoints and all of them failed, we will report only one of them, which might be unexpected. Maybe we should just wait for all of them? On the start of the system, all mailboxes should be empty, so I do not expect this to take long.